### DISCUSSION TITLE
favicon.ico not found on browsers

### DISCUSSION THREAD

**Discussion by carliedu** [clue/framework-x#232] [Q&A] [answered]
The file favicon.ico is never found.
After copying a ```favicon.ico``` file to the root, executing ```php public/index.php``` and calling http://myvps.com:8080/ on browser, I get the following in browsers console:
```GET
http://myvps.com:8080/favicon.ico
[HTTP/1.1 404 Not Found 0ms]
```

**Comment by SimonFrings** [comment, marked as answer]
Hey @carliedu, you're currently using the routing logic of Framework X to access a `favicon.ico`, you can read more about routing inside the documentation: https://framework-x.org/docs/api/app/#routing. This won't work by default (unless you have defined a specific routing behavior for that) but you can configure your web server of choice (Apache, nginx, etc.) in order give access to this file. 

FYI: I converted this ticket to a discussion as it seems to be more of a question instead of a bug report.

Hope this helps :+1:

**Comment by carliedu** [comment]
Hi @SimonFrings.
More and more, I feel like I'm the only one interested in using ReactPHP, and now Framework-X in browsers instead of the console. All usage instructions throughout the documentation refer to the console, making the calls with telnet or curl.
In console, favicon.ico is not used, but in Browsers it helps to identify the tab, when you have several open.
I opened this case as, to me, it looks like a bug since I didn't developed/changed anything. I used the ```index.php``` code exactly as I downloaded it from the ```examples``` folder. So the given example is "currently using the routing logic". I ran ```php public/index.php``` in my VPS console, but used Firefox for calling ```http://myvps.com:8080/```. This is where the Favicon error appears. I think this situation needs to be addressed. Sorry if I'm too detailed.
You also commented to use "configure your web server of choice (apache, nginx, etc.)", which I don't want. I really like the built-in web server.
What I'm trying to do, is promote Framework-X (and ReactPHP), but from my point of view (which is using Browsers and not console). I want to put together a tutorial, on how to use Framework-X to create a chat with a formatted screen.
Best

**Reply by SimonFrings** [reply]
> More and more, I feel like I'm the only one interested in using ReactPHP, and now Framework-X in browsers instead of the console.

@carliedu I don't think you're the only one as Framework X's purpose is to help creating powerful web APIs. We often refer to the console as a place to start the built-in webserver for local development and it additionally gives a lot of insights on incoming requests. You can also use the command line to test specific endpoints of your API with `curl`, but you should use your Browser as well in order to test the user's interaction with your API.

Our filesystem documentation is still in the works, but you can use the `FilesystemHandler` for this which is already part of Framework X. Although, the `FilesystemHandler` has been around for a long time now and we have plans to change this in the near future (create a new `FileHandler` and maybe get rid of the `FilesystemHandler`). I would still argue that it makes sense to use something like an Apache for this, but this is up to you.

(FYI: we also shared our roadmap for [Framework X on Twitter](https://twitter.com/cassifyit/status/1718888717023334414), just to give some insights on what we've planned for the future)
